trade-finance-manager-api/src/v1/controllers/deal.party-db.test.js

```diff
@@ -53,3 +56,165 @@ describe('getCompany returns company', () => {
     expect(result.data).toEqual(mockCompany);
   });
 });
+
+jest.mock('../api.js', () => ({
+  getPartyDbInfo: jest.fn(),
+  getOrCreatePartyDbInfo: jest.fn(),
+}));
+
+jest.mock('@ukef/dtfs2-common', () => ({
+  ...jest.requireActual('@ukef/dtfs2-common'),
+  isAutomaticSalesforceCustomerCreationFeatureFlagEnabled: jest.fn(),
+}));
+
+describe('when automatic Salesforce customer creation feature flag is disabled', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.mocked(isAutomaticSalesforceCustomerCreationFeatureFlagEnabled).mockReturnValue(false);
+  });
+
+  describe('getPartyUrn', () => {
+    it('should return an empty string if no companyRegNo is provided', async () => {
+      const result = await api.getPartyUrn({});
+      expect(result).toBe('');
+    });
+
+    it('should call getPartyDbInfo and return urn', async () => {
+      getPartyDbInfo.mockResolvedValue([{ partyUrn: 'TEST_URN' }]);
+
+      const companyData = { companyRegNo: '12345678' };
+
+      const result = await api.getPartyUrn(companyData);
+
+      expect(getPartyDbInfo).toHaveBeenCalledWith(companyData);
+      expect(getPartyDbInfo).toHaveBeenCalledTimes(1);
+
+      expect(result).toBe('TEST_URN');
+    });
+
+    it('should not call getOrCreatePartyDbInfo', async () => {
+      getPartyDbInfo.mockResolvedValue([{ partyUrn: 'TEST_URN' }]);
+
+      const companyData = { companyRegNo: '12345678' };
+
+      await api.getPartyUrn(companyData);
+
+      expect(getOrCreatePartyDbInfo).toHaveBeenCalledTimes(0);
+    });
+
+    it('should return an empty string if getPartyDbInfo returns false', async () => {
+      getPartyDbInfo.mockResolvedValue(false);
+
+      const companyData = { companyRegNo: '12345678' };
+
+      const result = await api.getPartyUrn(companyData);
+
+      expect(getPartyDbInfo).toHaveBeenCalledWith(companyData);
+      expect(getPartyDbInfo).toHaveBeenCalledTimes(1);
+
+      expect(result).toBe('');
+    });
+  });
+});
+
+describe('when automatic Salesforce customer creation feature flag is enabled', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.mocked(isAutomaticSalesforceCustomerCreationFeatureFlagEnabled).mockReturnValue(true);
+  });
+
+  it.each([
```

Comment:
would suggest declare the array as const and refer to the const as `it.each(invalidQueryParameter)...` please.

Reply:
Pending.

Reply:
Ah sorry, missed this one somehow- done now :)